### PR TITLE
chore: reduce indirection in DefaultTrackerBundleService

### PR DESCRIPTION
**Pull request by teleivo**
The pattern we often use of using a map of TrackerTypes to functions
introduces indirection making the code hard to read. We also rely on a
JAVA EE feature @PostConstruct to build the map once all dependencies
have been wired.

In this case the alternative code is straightforward to read. Removing
@PostConstruct makes unit testing also easier: no need for Spring doing
executing the @PostConstruct or us manually calling initMaps after the
constructor.

### To reviewers

I've seen this pattern at least a couple of times in tracker. It feels to smart for me 😅 I always have to re-read/think whats going on even though I've seen the pattern already. I would prefer a more boring, simple style at least in this case.

### CODE REVIEW

**Review by jbee — APPROVED** [review]
Generally like the readability of your version better. Don't think performance is a concern here.

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/DefaultTrackerBundleService.java [hunk 1]
```diff
@@ -149,10 +113,14 @@ public TrackerBundleReport commit( TrackerBundle bundle )
         }
 
         Session session = sessionFactory.getCurrentSession();
-
-        TrackerType.getOrderedByPriority()
-            .forEach( t -> bundleReport.getTypeReportMap().put( t, COMMIT_MAPPER.get( t )
-                .apply( session, bundle ) ) );
+        bundleReport.getTypeReportMap().put( TrackerType.TRACKED_ENTITY,
```
[the remark this answers is not in the document]
Comment (teleivo) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/DefaultTrackerBundleService.java:116]: I agree with a var for bundleReport.getTypeReportMap() 👍🏻  All the services are only used once so I would have to create one for each = 5. I think that would make it less readable.

**Comment by teleivo** [comment]
> 

I didn't mention performance 😋

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/DefaultTrackerBundleService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jbee) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/DefaultTrackerBundleService.java]: Haha, yes - missed that it were different persisters. Lambdas could still solve it but this is less of a cleanup we really should do.

**Review by enricocolasante — APPROVED** [review]

**Comment by teleivo** [comment]
@stian-sandvold could you maybe merge this with admin power 😄 . Semantic PR check does not answer

**Pull request merged**